app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/snippet/SnippetListDetailComposable.kt

```diff
@@ -0,0 +1,122 @@
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.compose.composable
+import com.bottlerocketstudios.brarchitecture.R
+import com.bottlerocketstudios.brarchitecture.ui.ComposeActivity
+import com.bottlerocketstudios.brarchitecture.ui.Routes
+import com.bottlerocketstudios.compose.snippets.CreateSnippetScreen
+import com.bottlerocketstudios.compose.snippets.SnippetUiModel
+import com.bottlerocketstudios.compose.snippets.SnippetsBrowserScreen
+import com.bottlerocketstudios.compose.snippets.SnippetsBrowserScreenState
+import com.bottlerocketstudios.compose.util.LaunchCollection
+import com.bottlerocketstudios.compose.util.asMutableState
+import com.bottlerocketstudios.compose.util.toStringIdHelper
+import com.bottlerocketstudios.compose.widgets.listdetail.AnimatedListDetail
+import kotlinx.coroutines.launch
+import org.koin.androidx.viewmodel.ext.android.getViewModel
+
+// Used to represent create item in list so it can be "selected" for detail view
+private val CreateSnippetItem = SnippetUiModel(
+    id = "CREATE_SNIPPET_SCREEN",
+    workspaceSlug = "",
+    title = "CREATE_SNIPPET_SCREEN",
+    userName = "UI_MODEL",
+    formattedLastUpdatedTime = "".toStringIdHelper()
+)
+
+// TODO - Custom Animations for entrance of detail.   Try to use same animation spec for navigation side and visibility
+fun ComposeActivity.newSnippetsComposable(navGraphBuilder: NavGraphBuilder) {
+    navGraphBuilder.composable(Routes.NewSnippet) {
+        val scope = rememberCoroutineScope()
+        val snippetsViewModel: SnippetsViewModel = getViewModel()
+        val config = LocalConfiguration.current
+
+        val list = snippetsViewModel.snippets.collectAsState(initial = emptyList())
+        controls.title = stringResource(id = R.string.snippets_title)
+        controls.topLevel = true
+
+        AnimatedListDetail(
+            list = list.value + CreateSnippetItem,
+            keyProvider = { it.id },
+            smallScreen = config.smallestScreenWidthDp < 580
+        ) {
+
+            // Define List UI and connect to VM
+            List { list, selected ->
+                // TODO - Use selected to highlight item
+                SnippetsBrowserScreen(
+                    state = SnippetsBrowserScreenState(
+                        // Don't display CreateSnippetPlaceHolder
+                        snippets = (list - CreateSnippetItem).asMutableState(),
+                        createVisible = snippetsViewModel.showCreateCTA.collectAsState(),
+                        onCreateSnippetClicked = {
+                            scope.launch {
+                                select("CREATE_SNIPPET_SCREEN")
+                            }
+                        }
+                    )
+                )
+            }
+
+            // Define Detail UI for create, detail, and empty states
+            Detail { model ->
+                model.also {
```

Comment:
@BottleRocket-Colin Should you be using `model?.also {` here since you handle null in the trailing `run` block? This would prevent null from being handled in both the `also` block and the trailing `run` block.

Reply:
The also block returns this (which is why this works right now), but you're correct in that we don't want to run the code in the empty else block.  No current bug, but would have been very strange when we added that section in.  Good spot!!